src/backend/distributed/planner/multi_explain.c

```diff
@@ -529,8 +530,31 @@ ExplainMapMergeJob(MapMergeJob *mapMergeJob, ExplainState *es)
 
 
 /*
- * ExplainTaskList shows the remote EXPLAIN for the first task in taskList,
- * or all tasks if citus.explain_all_tasks is on.
+ * CompareTasksByFetchedExplainAnalyzeDuration is a helper function to compare two tasks by their execution duration.
+ */
+static int
+CompareTasksByFetchedExplainAnalyzeDuration(const void *leftElement, const
+											void *rightElement)
+{
+	const Task *leftTask = *((const Task **) leftElement);
+	const Task *rightTask = *((const Task **) rightElement);
+
+	double leftTaskExecutionDuration = leftTask->fetchedExplainAnalyzeExecutionDuration;
+	double rightTaskExecutionDuration = rightTask->fetchedExplainAnalyzeExecutionDuration;
+
+	double diff = leftTaskExecutionDuration - rightTaskExecutionDuration;
+	if (diff != 0)
```

Comment:
Let's follow the structure that @SaitTalhaNisanci suggested and @marcocitus preferred

```C
if (diff > 0) {
return -1;
}else if ( diff < 0) {
return 1;
} else {
return 0;
}
```